bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/slider/BootstrapSlider.java

```diff
@@ -0,0 +1,258 @@
+            }
+        }
+
+        @Override
+        public String convertToString(ISliderValue value, Locale locale) {
+            return value.toString();
+        }
+    }
+    
+    private  IConverter<ISliderValue> converter;
+    
+    private N min;
+    private N max;
+    private N step;
+    private TooltipType tooltip;
+    private HandleType handle;
+    // function to create custom tooltip.
+    private String formatter;
+    private Orientation orientation;
+    
+    public BootstrapSlider(String id, IModel<T> model, Class<T> typeClass)
+    {
+        super(id, model, typeClass);
+        try {
+            if(Double.class.isAssignableFrom(typeClass.newInstance().getNumberType())) {
+                converter = new BootstrapDoubleSliderConverter();
+            } else if(Integer.class.isAssignableFrom(typeClass.newInstance().getNumberType())) {
+                converter = new BootstrapIntegerSliderConverter();
+            } else {
+                converter = new BootstrapLongSliderConverter();
+            }
+        } catch (InstantiationException e) {
+            throw new WicketRuntimeException(e);
+        } catch (IllegalAccessException e) {
+            throw new WicketRuntimeException(e);
+        }
+    }
+    
+    protected T newInstance() {
+        try {
+            return (T)getType().newInstance();
+        } catch (InstantiationException e) {
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public <C> IConverter<C> getConverter(final Class<C> type)
+    {
+        if (ISliderValue.class.isAssignableFrom(type))
+        {
+            return (IConverter<C>)converter;
+        }
+        else
+        {
+            return super.getConverter(type);
+        }
+    }
+
+    @Override
+    protected void onInitialize() {
+        super.onInitialize();
+        setOutputMarkupId(true);
+    }
+
+    @Override
+    protected void onComponentTag(ComponentTag tag) {
+        // Must be attached to an input tag
+        checkComponentTag(tag, "input");
+        tag.put("data-slider-value", converter.convertToString(getModelObject(), getLocale()));
+        tag.put("data-slider-min", min != null ? min.toString() : "0");
+        tag.put("data-slider-max",max != null? max.toString(): "10");
+        tag.put("data-slider-step",step != null? step.toString(): "1");
+        tag.put("type", "text");
+        if(tooltip != null) {
+            tag.put("data-slider-tooltip", tooltip.name());
+        }
+        if(handle != null) {
+            tag.put("data-slider-handle", handle.name());
+        }
+        if(!isEnabled()) {
+            tag.put("data-slider-enabled", false);
+        }
+        if(orientation != null) {
+            tag.put("data-slider-orientation", orientation.name());
+        }
+        tag.put("id", getMarkupId());
```

Comment:
Not it was on purpose... As I do not want all the logic on parent textField to be applied.